src/main/scala/castalia/matcher/MatcherActor.scala

```diff
@@ -0,0 +1,70 @@
+package castalia.matcher
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.http.scaladsl.model.HttpRequest
+import castalia.matcher.MatchResultGatherer.{MatchFound, MatchNotFound}
+import castalia.matcher.MatcherActor.RespondIfMatched
+import castalia.matcher.types.{Params, Segments}
+
+import scala.annotation.tailrec
+
+/**
+  * Created by m06f791 on 4-2-2016.
+  */
+object MatcherActor {
+
+  case class RespondIfMatched(parsedUri: ParsedUri, httpRequest: HttpRequest, gatherer: ActorRef)
+
+  def props(segments: Segments, handler: ActorRef): Props = Props(new MatcherActor(segments, handler))
+}
+
+class MatcherActor(segments: Segments, handler: ActorRef) extends Actor with ActorLogging {
+
+  def receive: Receive = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/scala-academy/Castalia/file/1993177519/issues/source?bid=2998213&fileBranchId=3120054#l23)